server/src/main/java/org/opensearch/gateway/PrimaryShardAllocator.java

```diff
@@ -381,22 +423,33 @@ protected static NodeShardsResult buildNodeShardsResult(
                         + nodeShardState.storeException();
                 numberOfAllocationsFound++;
                 if (matchAnyShard || inSyncAllocationIds.contains(nodeShardState.allocationId())) {
-                    nodeShardStates.add(nodeShardState);
+                    nodeShardStates.add(nodeShardState, nodeShardState.getNode());
                 }
             }
         }
 
+        if (logger.isTraceEnabled()) {
+            logger.trace(
+                "{} candidates for allocation: {}",
+                shard,
+                nodeShardStates.nodeShardStates.values().stream().map(DiscoveryNode::getName).collect(Collectors.joining(", "))
+            );
+        }
+        return new NodeShardsResult(nodeShardStates, numberOfAllocationsFound);
+    }
+
+    protected static Comparator<NodeShardState> getComparator(boolean matchAnyShard, Set<String> inSyncAllocationIds) {
```

Comment:
Can this method be private ? 

name suggestion - **createActiveShardComparator**

Reply:
Modified the name and made this function private.